### PR TITLE
feat(clouderror): treat http2.GoAwayError as Unavailable

### PR DESCRIPTION
**Pull request by thall**

### CODE REVIEW

**Comment by odsod** [comment]
@thall LGTM! anything to consider before merging?

File: clouderror/wrap_test.go [hunk 1]
```diff
@@ -70,6 +71,15 @@ func Test_WrapTransient(t *testing.T) {
 			err:          fmt.Errorf("bad: %w", context.Canceled),
 			expectedCode: codes.Canceled,
 		},
+		{
+			name: "http2.GoAwayError",
+			err: http2.GoAwayError{
```
[the remark this answers is not in the document]
Comment (thall) [reply clouderror/wrap_test.go:76]: I tested that, If I remember correctly it is thrown as value.
The switch case is true if the error is a value, not a pointer.
I can double check what is thrown.

[the remark this answers is not in the document]
Comment (thall) [reply clouderror/wrap_test.go:76]: Thrown as [value](https://github.com/golang/net/blob/65efbad9474a514a2f3c08716b8cf38011fa2736/http2/transport.go#L2215)
Here is a [test](https://github.com/golang/net/blob/master/http2/transport_test.go#L2911) as well

[the remark this answers is not in the document]
Comment (odsod) [reply clouderror/wrap_test.go:76]: Nice find! :detective: LGTM

**Pull request merged**